orderer/common/channelparticipation/restapi.go

```diff
@@ -51,7 +58,43 @@ func (h *HTTPHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 	}
 
 	switch req.Method {
-	case http.MethodGet, http.MethodPost, http.MethodDelete:
+	case http.MethodGet:
+		_, err := negotiateContentType(req) // Only application/json for now
+		if err != nil {
+			h.sendResponseJsonError(resp, http.StatusNotAcceptable, err)
+			return
+		}
+
+		channelID, all, err := detectResource(req.URL.Path)
+		if err != nil {
+			h.sendResponseJsonError(resp, http.StatusBadRequest, err)
+		}
+
+		// Asking for a list of all channels
+		if all {
+			var channels *ListAllChannels = &ListAllChannels{}
+			channels.Channels, channels.SystemChannel = h.registrar.ListAllChannels()
+			channels.Size = len(channels.Channels)
+			for i, info := range channels.Channels {
+				channels.Channels[i].URL = URLBaseV1Channels + `/` + info.Name
+			}
+			h.sendResponseOK(resp, channels)
+			return
+		}
+
+		// Asking for a single channel
+		if err = configtx.ValidateChannelID(channelID); err != nil {
+			h.sendResponseJsonError(resp, http.StatusBadRequest, errors.Wrap(err, "invalid channel ID"))
+			return
+		}
+		infoFull, err := h.registrar.ListChannel(channelID)
```

Comment:
I think there is a way to register different path patterns to distinguish the desired function call, that is to separate logic of list all channels versus retrieving specific channel information.

Reply:
True! using `github.com/gorilla/mux`, I'll use that.